src/FileType_XML.f90

```diff
@@ -55,19 +55,19 @@ MODULE FileType_XML
 !> Derived type for an XML element
 TYPE :: XMLElementType
   !> The number of attributes defined on the element
-  INTEGER(SIK),PRIVATE :: nAttr=0
```

Comment:
If they were set to private for some purpose (e.g., avoid accidentally overriding it), it should be better to write setter/getter to access them. Same thing applies to the below changes.